pkg/clouduploader/operation.go

```diff
@@ -146,3 +167,89 @@ func (c *Conn) ConnReadyWait() bool {
 	ok := <-*c.ConnReady
 	return ok
 }
+
+// listKeys return list of Keys -- files/directories
+// Note:
+// - list may contain incomplete list of keys, check for error before using list
+// - listKeys uses '/' as delimiter.
+func (c *Conn) listKeys(prefix string, keyType int) ([]string, error) {
+	keys := []string{}
+
+	lister := c.bucket.List(&blob.ListOptions{
+		Delimiter: "/",
+		Prefix:    prefix,
+	})
+	for {
+		obj, err := lister.Next(c.ctx)
+		if err == io.EOF {
+			break
+		}
+
+		if err != nil {
+			c.Log.Errorf("Failed to get next blob err=%v", err)
+			return keys, err
+		}
+
+		switch keyType {
+		case KeyBoth:
+		case KeyFile:
+			if obj.IsDir {
+				continue
+			}
+		case KeyDirectory:
+			if !obj.IsDir {
+				continue
+			}
+		default:
+			c.Log.Warningf("Invalid keyType=%d, Ignored", keyType)
```

Comment:
should there be a continue here to ignore?

Reply:
yes